jax/experimental/optimizers.py

```diff
@@ -478,7 +478,7 @@ def broadcast_into(ndim, x, axis):
     return x[tuple(idx)]
 
   def init(x0):
-    vs = [jnp.zeros(sz, dtype=x0.dtype) for sz in x0.shape]
+    vs = [jnp.zeros(sz, dtype=x0.dtype) for sz in (x0+jnp.zeros(1)).shape]
```

Comment:
It looks like your intent here is to ensure that `x0` is at least 1 dimensional. With that in mind, I think using `jnp.atleast1d(x0)` would be a better approach.

Reply:
Thank you, I learned a new function today. I've updated the PR.